framework/src/actions/SetupMeshAction.C

```diff
@@ -243,9 +251,18 @@ SetupMeshAction::act()
         }
         else if (!_moose_object_pars.get<bool>("_mesh_generator_mesh"))
         {
-          mooseWarning("Mesh Generators present but the [Mesh] block is set to construct a \"",
-                       _type,
-                       "\" mesh, which does not use Mesh Generators in constructing the mesh.");
+          // There are cases where a custom action may register the "add_mesh_generator" task, but
+          // may not actually add any mesh generators depending on user input. We don't want to risk
+          // giving false warnings in this case. However, if we triggered the "add_mesh_generator"
+          // task through explicit input file syntax, then it is definitely safe to warn
+          for (auto generator_action_ptr : generator_actions)
+            if (dynamic_cast<AddMeshGeneratorAction *>(generator_action_ptr))
```

Comment:
I don't love this, but I don't have a better solution right now. This all stems from maintaining our legacy default "types" for Meshes and adding more. I guess this is ok.